pandas/tests/base/test_misc.py

```diff
@@ -147,19 +147,29 @@ def test_searchsorted(request, index_or_series_obj):
     # See gh-12238
     obj = index_or_series_obj
 
+    # 1. Check for multi-index
     if isinstance(obj, pd.MultiIndex):
-        # See gh-14833
-        request.applymarker(
-            pytest.mark.xfail(
-                reason="np.searchsorted doesn't work on pd.MultiIndex: GH 14833"
-            )
-        )
-    elif obj.dtype.kind == "c" and isinstance(obj, Index):
-        # TODO: Should Series cases also raise? Looks like they use numpy
-        #  comparison semantics https://github.com/numpy/numpy/issues/15981
-        mark = pytest.mark.xfail(reason="complex objects are not comparable")
-        request.applymarker(mark)
-
+        request.applymarker(pytest.mark.xfail(reason="GH 14833", strict=False))
+        return
+
+    # 2. Check for Index and subtypes
+    if isinstance(obj, Index):
+        # 2a. Mixed types
+        if obj.inferred_type in ["mixed", "mixed-integer"]:
+            try:
+                obj = obj.astype(str)
+            except (TypeError, ValueError):
+                request.applymarker(
+                    pytest.mark.xfail(reason="Mixed types", strict=False)
+                )
+                return
+
+        # 2b. Complex types
+        elif obj.dtype.kind == "c":
+            request.applymarker(pytest.mark.xfail(reason="Complex types", strict=False))
```

Comment:
```suggestion
                    pytest.mark.xfail(reason="Mixed types")
                )
                return

        # 2b. Complex types
        elif obj.dtype.kind == "c":
            request.applymarker(pytest.mark.xfail(reason="Complex types"))
```

We don't use `strict=False` generally